safeweb/http.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package safeweb provides a simple, safe, and opinionated wrapper around
+// http.Server that applies some web application security defenses by default.
+// It is intended to be used in place of http.Server for internal applications
+// that reside on the Tailscale network.
+//
+// safeweb requires that applications be built using two distinct
+// http.ServeMuxes for serving browser and API clients to serve the appropriate
+// security defenses in their respective contexts. When serving requests it
+// will first attempt to route requests using the browser mux, and if no
+// matching route is found it will instead defer to the API mux.
+//
+// safeweb enforces Cross-Site Request Forgery (CSRF) protection for all routes
```

Comment:
Maybe two headings?

```go
// # Browser Routes
//
// All routes in the browser mux enforce CSRF protection using the gorilla/csrf
// package. The application must template the CSRF token into its forms using the
// [TemplateField] and [TemplateTag] APIs. Applications that are served over HTTPS
// should also set the SecureContext field to true, ensuring that CSRF cookies are
// marked as Secure.
//
// In addition, browser routes will also receive:
//
//   - A Content-Security-Policy header that disallows inline scripts, framing, and third-party resources.
//   - An X-Content-Type-Options header on responses set to "nosniff" to prevent MIME type sniffing attacks.
//
// [TemplateField]: https://godoc.org/github.com/gorilla/csrf#TemplateField
// [TemplateTag]: https://godoc.org/github.com/gorilla/csrf#TemplateTag
//
// # API Routes
//
// Routes in the API mux enforce important CORS rules. The server validates the
// Content-Type header to block `application/x-www-form-urlencoded` requests, and
// adds CORS headers if they are provided in the configuration.
```

This link notation will let the links be clicky when they show up in godoc. And the headers are just a nice visual separation, understood by the godoc renderer.